app/src/main/java/com/google/samples/apps/sunflower/GardenFragment.kt

```diff
@@ -20,52 +20,45 @@ import android.arch.lifecycle.Observer
 import android.view.LayoutInflater
-import android.view.View
+import android.view.View.GONE
 import android.view.ViewGroup
-import android.widget.TextView
 import com.google.samples.apps.sunflower.adapters.GardenPlantingAdapter
+import com.google.samples.apps.sunflower.databinding.FragmentGardenBinding
 import com.google.samples.apps.sunflower.utilities.InjectorUtils
 import com.google.samples.apps.sunflower.viewmodels.GardenPlantingListViewModel
 
 class GardenFragment : Fragment() {
 
     override fun onCreateView(
-            inflater: LayoutInflater,
-            container: ViewGroup?,
-            savedInstanceState: Bundle?
-    ): View? {
-        val view = inflater.inflate(R.layout.fragment_garden, container, false)
-        val adapter = GardenPlantingAdapter(view.context)
-        view.findViewById<RecyclerView>(R.id.garden_list).adapter = adapter
-        subscribeUi(adapter)
-        return view
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ) = FragmentGardenBinding.inflate(inflater, container, false).run {
+        setLifecycleOwner(this@GardenFragment)
+        val context = context ?: return root
+        with(GardenPlantingAdapter(context)) {
+            gardenList.adapter = this
+            subscribeUi(this@run, this)
+        }
+        root
     }
 
-    private fun subscribeUi(adapter: GardenPlantingAdapter) {
+    private fun subscribeUi(databinding: FragmentGardenBinding, adapter: GardenPlantingAdapter) {
         val factory = InjectorUtils.provideGardenPlantingListViewModelFactory(requireContext())
-        val viewModel = ViewModelProviders.of(this, factory)
-                .get(GardenPlantingListViewModel::class.java)
+        val viewModel =
+            ViewModelProviders.of(this, factory).get(GardenPlantingListViewModel::class.java)
 
         viewModel.getGardenPlantings().observe(this, Observer { plantings ->
-            if (plantings != null && plantings.isNotEmpty()) {
-                activity?.run {
-                    findViewById<RecyclerView>(R.id.garden_list).run { visibility = View.VISIBLE }
-                    findViewById<TextView>(R.id.empty_garden).run { visibility = View.GONE }
-                }
-            } else {
-                activity?.run {
-                    findViewById<RecyclerView>(R.id.garden_list).run { visibility = View.GONE }
-                    findViewById<TextView>(R.id.empty_garden).run { visibility = View.VISIBLE }
-                }
-            }
+            databinding.setVariable(BR.hasPlantings, plantings != null && plantings.isNotEmpty())
         })
 
         // TODO handle this better - shouldn't need to filter
         viewModel.getPlantAndGardenPlantings().observe(this, Observer { result ->
             if (result != null && result.isNotEmpty())
                 adapter.values = result.filter { it.gardenPlantings.isNotEmpty() }
+
+            databinding.loadingUi.visibility = GONE
```

Comment:
I think that you could remove static import as you did in `PlantListFragment.kt`

Reply:
@piotrek1543  Thx for this point, have corrected.